### DISCUSSION TITLE
How to retrieve edited values of nearest cells, not initial values

### DISCUSSION THREAD

**Discussion by asifka-ramz** [KevinVandy/material-react-table#838] [Q&A]
Hello,

I hope this message finds you well. I'm currently working on a table editing feature where I need some guidance.

In this feature, while editing a cell value, I'm facing an issue with validating the values in the nearest cells. Specifically, I'm using the onBlur or onChange event, but I'm only able to retrieve the initial values of adjacent cells and not the edited values.

Has anyone encountered a similar challenge or have insights on how to address this issue? Your help would be greatly appreciated!

Thank you in advance.

**Comment by asifka-ramz** [comment]
Could anybody help with how to retrieve values from adjacent cells for validation?

**Reply by KevinVandy** [reply]
row._valueCache. You should be able to find this with some console logs of the rows

**Reply by asifka-ramz** [reply]
I have tried using row._valueCache, but it only retrieves the last edited cell's value correctly. The values of the adjacent cells remain the default values. Although the edited values are displayed in the table, I'm encountering difficulties retrieving them in the onBlur event.  which state is currently used to show the edited values in the table cells?

Thanks

**Comment by asifka-ramz** [comment]
@KevinVandy , could you please reply? It would be very useful. I am currently stuck with this, and many others will find it helpful. Sorry for annoying you.

Thanks
